duktape/src/main/java/com/squareup/duktape/Duktape.java

```diff
@@ -63,6 +65,30 @@ public String evaluate(String script) {
   }
 
   /**
+   * Binds {@code object} to {@code name} for use in JavaScript as a global object.  {@type} defines
+   * the interface implemented by {@code object} that will be accessible to JavaScript.
+   */
+  public <T> void bind(Class<T> type, String name, T object) {
+    if (!type.isInterface()) {
+      throw new UnsupportedOperationException("Only interfaces can be bound.");
+    }
+    if (type.getInterfaces().length > 0) {
+      throw new UnsupportedOperationException("Bound interfaces cannot extend other interfaces.");
+    }
+    if (!type.isInstance(object)) {
+      throw new IllegalArgumentException("object is not an instance of the given type.");
+    }
+    HashMap<String, Method> methods = new HashMap<>();
```

Comment:
Always LinkedHashMap. Never HashMap. I even wrote a long story about why.
https://publicobject.com/2016/02/08/linkedhashmap-is-always-better-than-hashmap/